packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -5681,6 +5704,22 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
         ),
       ),
     );
+
+    // If this is a nested Navigator, handle system backs here so that the root
+    // Navigator doesn't get all of them.
+    if (widget.handlesBacksWhenNested && Navigator.maybeOf(context, rootNavigator: true) != this) {
+      return PopScope(
+        canPop: !canPop(),
```

Comment:
what would trigger the rebuild when canpop changes?

Reply:
Updated to trigger a rebuild.